boot-core/src/daemon/contract.rs

```diff
@@ -89,28 +103,48 @@ where
         Ok(wasm_code_path)
     }
 
-    /// Calculate the checksum of the wasm file to compare against previous uploads
+    // Calculate the checksum of the wasm file to compare against previous uploads
+    // NOTE: Should we change this? (and maybe move it out of here?, perhaps to an utils file or something)
+    // if this function at the bottom returns the checksum of the file
+    // when the checksums.txt file is not found what is the difference?
+    // what i mean is why is relevant here to check for the existence of a checksums.txt file
+    // parse it and find our contract in it
+    // if the default way is to do it in the function anyway
+    // for example: what if the file is named *checksums_intermediate.txt*
+    // which is the default name rust-optimizer generates?
```

Comment:
checksums.txt is the default name for workspace optimizer.

Reply:
Hmmm i get your point. Could you verify that the hash of the digest is actually equal? if it is then sure we can remove that code! Hashing is a bit expensive but we don't care that much about performance.

Reply:
1) workspace optimizer and the rust optimizer have a different output filename?
https://github.com/CosmWasm/rust-optimizer/blob/main/optimize.sh
I had no clue

2) I already did some testing and asked GPT4 about it (for a quick answer) hehe